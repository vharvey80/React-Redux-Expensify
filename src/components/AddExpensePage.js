import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';

import { addExpense } from '../actions/expenses.action';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.addExpense(expense);
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <h3>Add Expense</h3>
        <ExpenseForm 
          onSubmit={this.onSubmit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
