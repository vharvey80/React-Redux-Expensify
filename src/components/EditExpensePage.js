import React from 'react';
import { connect } from 'react-redux';

import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses.action';

export class EditExpensePage extends React.Component {
        onSubmit = (expense) => {
                this.props.editExpense(this.props.expense.id, expense);
                this.props.history.push('/');
        };

        onRemove = () => {
                this.props.removeExpense({ id: this.props.expense.id });
                this.props.history.push('/');
        };

        render() {
                return (
                        <div>
                                <ExpenseForm 
                                        onSubmit={this.onSubmit}
                                        expense={this.props.expense}
                                />
                                <button onClick={this.onRemove}>Remove Expense</button>
                        </div>                          
                );
        }
}

const mapDispatchToProps = (dispatch, props) => ({
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: (id) => dispatch(removeExpense(id))          
});

const mapStateToProps = (state, props) => ({
        expense: state.expenses.find((expense) => expense.id === props.match.params.id)      
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);