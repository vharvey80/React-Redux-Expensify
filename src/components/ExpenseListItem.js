import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <div>
        <hr />
        <Link to={`/edit/${id}`}>
            <h4>{description}</h4>
        </Link>
        <p>
            {numeral(amount/100).format('$0,0.00')} 
             -  
            {moment(createdAt).format('MMM Do, YYYY')}
        </p>
    </div>
);

export default ExpenseListItem;