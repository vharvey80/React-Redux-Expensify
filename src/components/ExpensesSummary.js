import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

import getExpensesTotal from '../helpers/expense-total.helper';
import getVisibleExpenses from '../helpers/expenses.helper';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">
                    Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span>
                </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getExpensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);