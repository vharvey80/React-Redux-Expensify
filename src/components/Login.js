import React from 'react';
import { connect } from 'react-redux';

import { startLogin } from '../actions/auth.action';

export const Login = ({ startLogin }) => (
  <div>
    <h2>Login</h2>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);