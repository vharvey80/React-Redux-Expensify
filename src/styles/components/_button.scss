.button {
    background: lighten($color: $dark-blue, $amount: 10%);
    border: none;
    border-radius: 3px;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-large;
    line-height: 1;
    padding: $s-size;
    text-decoration: none;
}

.button--link {
    background: none;
    font-family: Roboto;
}

.button--secondary {
    background: #888;
}