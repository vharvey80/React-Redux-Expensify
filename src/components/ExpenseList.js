import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';

import filterExpenses from '../helpers/expenses.helper';

export const ExpenseList = ({ expenses }) => (
    <div>
        {
            expenses.length === 0
                ? <p>No expenses</p>
                : <div>
                    {expenses.map((expense, index) => <ExpenseListItem key={expense.id} {...expense} />)}
                  </div>
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: filterExpenses(state.expenses, state.filters)
    };
}

export default connect(mapStateToProps)(ExpenseList);

