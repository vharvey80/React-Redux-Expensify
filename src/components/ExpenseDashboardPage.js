import React from 'react';

import ExpensesList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpensesList />
  </div>
);

export default ExpenseDashboardPage;
