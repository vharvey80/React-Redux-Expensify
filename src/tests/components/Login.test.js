import React from 'react';
import { shallow } from 'enzyme';

import { Login } from '../../components/Login';

let startLoginSpy, wrapper;

beforeEach(() => {
    startLoginSpy = jest.fn();
    wrapper = shallow(<Login startLogin={startLoginSpy} />);
});

test('Should render Login correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('Should call startLogin on button click.', () => {
    wrapper.find('button').simulate('click');

    expect(startLoginSpy).toHaveBeenCalled();
});