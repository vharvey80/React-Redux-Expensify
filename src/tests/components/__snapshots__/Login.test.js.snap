// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Login correctly 1`] = `
<div>
  <h2>
    Login
  </h2>
  <button
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
